homeassistant/components/switch_as_x/entity.py

```diff
@@ -10,32 +10,52 @@
     SERVICE_TURN_ON,
     STATE_ON,
     STATE_UNAVAILABLE,
-    EntityCategory,
 )
-from homeassistant.core import Event, callback
-from homeassistant.helpers import entity_registry as er
-from homeassistant.helpers.entity import Entity, ToggleEntity
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.entity import DeviceInfo, Entity, ToggleEntity
 from homeassistant.helpers.event import async_track_state_change_event
 
 from .const import DOMAIN as SWITCH_AS_X_DOMAIN
 
 
 class BaseEntity(Entity):
-    """Represents a Switch as a X."""
+    """Represents a Switch as an X."""
 
     _attr_should_poll = False
 
     def __init__(
         self,
-        name: str,
+        hass: HomeAssistant,
+        config_entry_title: str,
```

Comment:
Technically this is dynamic too

Reply:
Not really, `switch_as_x` config entries are hidden from the user